internal/framework6provider/float64_precision_resource_test.go

```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("framework_float64_precision.test", "float64_attribute", "340282346638528859811704183484516925440"),
+				),
+				ExpectError: regexp.MustCompile(`.*planned value\s{0,10}cty\.NumberIntVal\(3\.4028234663852886e\+38\) does not match config value\s{0,10}cty\.NumberIntVal\(3\.4028234663852885981170418348451692544e\+38\)`),
+			},
+			// No error when planned with 53-bit precision
+			{
+				Config: `resource "framework_float64_precision" "test" {
+					float64_attribute = 340282346638528860000000000000000000000
+				}`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("framework_float64_precision.test", "float64_attribute", "340282346638528860000000000000000000000"),
+				),
+			},
+			// Semantic equality with 53-bit precision scientific notation representation
+			{
+				Config: `resource "framework_float64_precision" "test" {
+					float64_attribute = 3.4028234663852886e+38
+				}`,
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("framework_float64_precision.test", "float64_attribute", "340282346638528860000000000000000000000"),
+				),
+			},
+			// No plan is detected when 53-bit precision value is replaced with higher precision value
+			{
+				Config: `resource "framework_float64_precision" "test" {
+					float64_attribute = 3.40282346638528859811704183484516925440e+38
+				}`,
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectEmptyPlan(),
+					},
+				},
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("framework_float64_precision.test", "float64_attribute", "340282346638528860000000000000000000000"),
+				),
+			},
+		},
+	})
+}
+
+// This test documents edge case behavior with math.MaxFloat32 with Terraform v1.9.0 and above
+// This number is unique in that it has an exact representation in both float32 and float64,
+// it is also an integer, which may affect how this value gets encoded/decoded with go-cty.
+// https://github.com/hashicorp/terraform-plugin-framework/issues/1017
+//
+// go-cty v1.14.4 uses string msgpack encoding instead of float msgpack encoding for large whole numbers
+// https://github.com/hashicorp/terraform/commit/cd252557e4200e031b942d9ff35c455bb30d858f
+// This changes allows the math.MaxFloat32 value to succeed in planning but fail during apply.
+// Terraform v1.9.0-alpha20240501 is the first Terraform version to use this updated encoding.
+func TestSchemaResource_Float64Attribute_Precision_MaxFloat32_TF1_9(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(version.Must(version.NewVersion("1.9.0-alpha20240501"))),
```

Comment:
Nit: If we can, let's just use the stable version for this `v1.9.0`

Similar note for the `framework5provider` equivalent.